src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAOSpec.scala

```diff
@@ -63,4 +65,140 @@ class HttpGoogleServicesDAOSpec extends FlatSpec with Matchers {
     objectMetadata.name should equal("test-composite-object")
     objectMetadata.md5Hash should equal(None)
   }
+
+  {
+    val headers = List[AnyRef]("header 1", "header 2", "header 3").asJava
+
+    val row1 = List[AnyRef]("proj 1", "free", "trial").asJava
+    val row2 = List[AnyRef]("proj 2", "firecloud", "").asJava
+    val row3 = List[AnyRef]("proj 3", null, "data").asJava
+
+    val cells1 = List(headers, row1, row2, row3)
+
+    it should "have no changes if the data didn't change" in {
+      check(
+        newContent = cells1,
+        existingContent = cells1,
+        expectedOutput = cells1
+      )
+    }
+
+    it should "not explode if there is no data (only headers)" in {
+      check(
+        newContent = List(headers),
+        existingContent = List(headers),
+        expectedOutput = List(headers)
+      )
+    }
+
+    // This happens when calling updateBillingReport the first time from within createBillingReport
```

Comment:
This is great and makes me think of another edge case: 
What would happen if `updateBillingReport` or `createBillingReport` were called when there is no data yet. If this is a valid scenario, it should handle when newContent==null and existingContent==null (or is an empty list)

Reply:
agree this is a valid edge case, but wonder if it's worth expending effort on. Remember that very few people - only campaign managers, and then a subset of those - will call this, so we can be pretty tolerant of unexpected errors.

Reply:
It's possible, but I'm comfortable leaving the behavior undefined, a.k.a. "don't do that"